packages/gatsby/src/utils/handle-flags.ts

```diff
@@ -16,11 +18,54 @@ const handleFlags = (
   flags: Array<IFlag>,
   configFlags: Record<string, boolean>,
   executingCommand = process.env.gatsby_executing_command
-): { enabledConfigFlags: Array<IFlag>; message: string } => {
+): {
+  enabledConfigFlags: Array<IFlag>
+  unknownFlagMessage: string
+  message: string
+} => {
   // Prepare config flags.
   // Filter out any flags that are set to false.
   const availableFlags = new Map()
   flags.forEach(flag => availableFlags.set(flag.name, flag))
+
+  // Find unknown flags someone has in their config to warn them about.
+  const unknownConfigFlags = Object.keys(configFlags)
+    .filter(flagName => !availableFlags.has(flagName))
+    .map(flag => {
+      const flagsWithDistance = flags.map(f => {
+        return {
+          name: f.name,
+          distance: distance(flag, f.name),
+        }
+      })
+
+      const minDistance = _.minBy(flagsWithDistance, f => f.distance)
+
+      let didYouMean
+      if (minDistance) {
+        didYouMean = minDistance.distance < 4 ? minDistance.name : undefined
+      }
+
+      return {
+        flag,
+        didYouMean,
+      }
+    })
```

Comment:
I know functional programming is nice, but it's really bad performance in JS (at scale) with extra loops for things that can be done in 1 loop and extra GC churn for all those temporary arrays - we should avoid them. It doesn't matter that much here because number of flags will be low, but if we use those patterns a lot, then others will try to use where it actually does matter too ;) (👋   @pvdz)

below might not work exactly - just something that should be same as above but it drops couple of loops and teporary array creations that need to be GCed
```suggestion
  const unknownConfigFlags = []
  for (let flagName in configFlags) {
    if (!availableFlags.has(flagName)) {
      return
    }
    let flagWithMinDistance, minDistance
    for (let availableFlag of flags) {
      const distanceToFlag = distance(flagName, availableFlag.name)
      if (!flagWithMinDistance || distanceToFlag < minDistance) {
        flagWithMinDistance = availableFlag.name
        minDistance = distanceToFlag
      }
    }
    
    unknownConfigFlags.push({
      flag: flagName,
      didYouMean: minDistance < 4 ? flagWithMinDistance : undefined
    })
  }
```